src/Perpetuum/Services/EventServices/EventListenerService.cs

```diff
@@ -47,12 +47,14 @@ public void AttachListener(EventProcessor<EventMessage> observer)
             _observers.Add(observer);
         }
 
-
         public override void Update(TimeSpan time)
         {
-            if (_queue.TryDequeue(out var message))
+            while (!_queue.IsEmpty)
```

Comment:
This could lead to an infinite loop without the main game loop updating. It might be worth time-gating this as well (within the expected server tickrate) so that everything left will be updated next "frame".

Reply:
I was concerned about this too!  Wasn't sure the cleanest way to handle it (like process a fixed maximum of messages, and let the next update try to chew on the rest).
I like the time-gate as that ensures 2 update calls will not overlap (+/-).

I'll work that in, thanks! =D